sdk/python/kfmd/metadata.py

```diff
@@ -114,22 +113,74 @@ def __init__(self, workspace=None, name=None, description=None):
       workspace {Worspace} -- Requried workspace object.
       name {str} -- Requried name of this run.
       description {str} -- Optional description.
+    """
+    # TODO(zhenghuiwang): check each field's type and whether set.
+    self.workspace = workspace
+    self.name = name
+    self.description = description
+
+class Execution(object):
+  """
+  Captures a run of pipeline or notebooks in a workspace and group executions.
+  """
+  EXECUTION_TYPE_NAME = "kubeflow.org/alpha/execution"
 
-    Creates a new run in a workspace and an execution for this run.
-    The run.log_XXX() methods will attach corresponding artifacts as the
+  def __init__(self, name=None, workspace=None, run=None, description=None):
+    """
+    Args:
+      name {str} -- Requried name of this run.
+      workspace {Worspace} -- required workspace object.
+      run {Run} -- optional run object.
+      description {str} -- Optional description.
+
+    Creates a new execution in a workspace and run.
+    The exection.log_XXX() methods will attach corresponding artifacts as the
     input or output of this execution.
 
-    Returns a run object for logging.
+    Returns an execution object for logging.
     """
     # TODO(zhenghuiwang): check each field's type and whether set.
-    self.workspace = workspace
+    self.id = None
     self.name = name
+    if workspace == None:
+      raise ValueError("'workspace' must be set.")
+    self.workspace = workspace
+    self.run = run
     self.description = description
+    self.create_time = get_rfc3339_time()
+    response = self.workspace._client.create_execution(
+        parent=self.EXECUTION_TYPE_NAME,
+        body=self.serialization(),
+    )
+    self.id = response.execution.id
 
-  def log(self, artifact):
+  def serialization(self):
```

Comment:
`serialized()` or `as_serialized_execution`or `as_mlmd_execution` or something along those lines. This returns a proto I assume?